src/LinearSolvers/preconditioners.jl

```diff
@@ -23,22 +24,45 @@ Creates an object for the block-Jacobi preconditioner
 * `cupart`: `part` transferred to the GPU
 * `P`: The sparse precondition matrix whose values are updated at each iteration
 """
-mutable struct BlockJacobiPreconditioner <: AbstractPreconditioner
-    npart::Int64
-    nJs::Int64
-    partitions::Vector{Vector{Int64}}
-    cupartitions
-    Js
-    cuJs
-    map
-    cumap
-    part
-    cupart
-    P
-    id
-    function BlockJacobiPreconditioner(J, npart, device=KA.CPU())
-        if isa(J, CUSPARSE.CuSparseMatrixCSR)
+struct BlockJacobiPreconditioner{AT,GAT,VI,GVI,MT,GMT,MI,GMI,SMT} <: AbstractPreconditioner
+    nblocks::Int64
```

Comment:
The best practice is to use `Int` instead of `Int64` (for 32 bits architecture)

Reply:
I spent too much time in code converting `int` to `long long int` due to overflows that I tend to be on the cautious side. In fact, I don't see a reason for not being explicit. The code would still work on a 32bit architecture. And if it's indeed a performance issue we should explicitly have an Integer type.

Reply:
Which we could actually also do for the `Float64`, especially on the GPU.